File: python/sprint2/D/code.py
# Comment it before submitting
# class Node:  
#     def __init__(self, value, next_item=None):  
#         self.value = value  
#         self.next_item = next_item

def solution(node, elem):
    # Your code
    # ヽ(´▽`)/
    pass

def test():
    node3 = Node("node3", None)
    node2 = Node("node2", node3)
    node1 = Node("node1", node2)
    node0 = Node("node0", node1)
    idx = solution(node0, "node2")
    # result is idx == 2

File: cpp/sprint2/D/code.cpp
// ! Remove define before submitting !
#define local

#ifndef local
#include "solution.h"
#endif

#ifdef local
#include <cassert>
#include <string>
#endif



#ifdef local

struct Node {  
    std::string value;  
    Node* next;  
    Node(const std::string &value, Node* next) : value(value), next(next) {
        
    }
};
#endif


int solution(Node* head, const std::string& elem) {
    // Your code
    // ヽ(´▽`)/
}

#ifdef local
void test() {
    Node node3("node3", nullptr);
    Node node2("node2", &node3);
    Node node1("node1", &node2);
    Node node0("node0", &node1);
    int idx = solution(&node0, "node2");
    assert(idx == 2);
}

int main() {
    test();
}
#endif